app/src/pages/Lobby/Inputs.tsx

```diff
@@ -163,13 +154,9 @@ export function SubmissionsPerPlayerInput(props: { value: string }) {
   const currentPlayer = React.useContext(CurrentPlayerContext)
   const currentGame = React.useContext(CurrentGameContext)
   const [updateGameSettings] = useUpdateGameSettingsMutation()
-  const [textFieldValue, setTextFieldValue] = React.useState("")
+  const [textFieldValue, setTextFieldValue] = React.useState(props.value)
   const canConfigureSettings = currentPlayer.role === PlayerRole.Host
 
-  React.useEffect(() => {
-    setTextFieldValue(props.value)
-  }, [props.value])
-
```

Comment:
@wtachau I don't believe these are equivalent

in the new version...the value from props get set once, upon the component mounting

in the old version... every time props change, it'll overwrite the state value. So when we fire the mutation, we update the db, which then causes the subscription in the parent component to receive that data and propogate through props again. That new value (flowing thrugh props) is needed for the observers (non-host participants) who will get their settings refreshed. For the host, it won't matter much bc it should be excactly what they typed (via state). (this is not a perfect solution either - there are some edge cases when it flashes to a previous state then back)

does that make sense?

Reply:
@avimoondra ah, I misunderstood what was going on here. I thought the point of this hook was to set the `textFieldValue` upon the first render, and on subsequent updates it would get updated by the `onChange` prop below (which calls `setTextFieldValue`). I thought this was equivalent because we were just setting the initial value with `React.useState(props.value)`.

I didn't think about the case where the value is updated remotely via the host, and we want it to update for the observers as well. Will update 👍